### PR TITLE
MWPW-158389: aem client updates for content-navigation wc

### PR DESCRIPTION
**Pull request by yesil**
This PR contains aem.js and related logic update to support
content-navigation web component in M@S studio.

Also, Milo spectrum build logic is updated to make all the transitive imports relative instead of absolute.
This change is needed in order to test better merch-card with sp-button for CCD use cases.
By this occasion, I also bumped the spectrum version to latest stable.

Resolves: [MWPW-158389](https://jira.corp.adobe.com/browse/MWPW-158389)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/ilyas/merch-card?martech=off
- After: https://content-navigation--milo--adobecom.hlx.page/drafts/ilyas/merch-card?martech=off

CC Pages:
- https://main--cc--adobecom.hlx.live/products/catalog?milolibs=content-navigation
- https://main--cc--adobecom.hlx.live/creativecloud?milolibs=content-navigation

### CODE REVIEW

File: libs/features/mas/web-components/src/variants/special-offer.js [hunk 1]
```diff
@@ -42,6 +42,10 @@ export class SpecialOffer extends VariantLayout {
     :host([variant='special-offers']) {
       min-height: 439px;
     }
+
+    :host([variant='special-offers']) {
```
Comment (yesil) [inline libs/features/mas/web-components/src/variants/special-offer.js:46]: this change is needed to render a merch-card with valid dimension without the responsive grid.

**Review by Roycethan — CHANGES_REQUESTED** [review]
@yesil 
1)The CTA alignment on Mini-compare cards are disturbed
https://main--cc--adobecom.hlx.live/creativecloud?milolibs=content-navigation--milo--adobecom
image
https://main--dc--adobecom.hlx.live/acrobat/features?milolibs=content-navigation--milo--adobecom
image

2) Also Possible to test this feature https://jira.corp.adobe.com/browse/MWPW-158389 if possible plz provide instructions thanks

cc @npeltier @afmicka

File: libs/features/mas/web-components/src/merch-card.js [hunk 2]
```diff
@@ -102,22 +105,35 @@ export class MerchCard extends LitElement {
...
-            // Filter out prices within the callout-content slot
-            const prices = allPrices.filter(
-                (price) => !price.closest('[slot="callout-content"]'),
-            );
-            await Promise.all(prices.map((price) => price.onceSettled()));
-            this.variantLayout.postCardUpdateHook(this);
-        });
+        if (changedProperties.has('variant')) {
+            this.variantLayout = getVariantLayout(this);
+            this.variantLayout.connectedCallbackHook();
```
Comment (npeltier) [inline libs/features/mas/web-components/src/merch-card.js:110]: either we rename that method, either we don't use it in constructor

File: libs/features/mas/web-components/src/merch-card.js [hunk 3]
```diff
@@ -222,13 +240,11 @@ export class MerchCard extends LitElement {
...
 
     get startingAt() {
-      return this.classList.contains('starting-at');
+        return this.classList.contains('starting-at');
     }
 
     connectedCallback() {
         super.connectedCallback();
-        this.variantLayout = getVariantLayout(this);
-        this.variantLayout.connectedCallbackHook();
```
Comment (npeltier) [inline libs/features/mas/web-components/src/merch-card.js:231]: i guess then it should just be renamed

File: libs/features/mas/web-components/src/merch-card.js (hunk 2 above)
Comment (npeltier) [reply libs/features/mas/web-components/src/merch-card.js:110]: or may be we can move connectedCallbackHook in some internal method of VariantLayout constructor?

File: libs/features/mas/web-components/src/variants/variant-layout.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ export class VariantLayout {
 
   constructor(card) {
     this.card = card;
-    setTimeout(() => this.insertVariantStyle(), 1);
+    this.insertVariantStyle();
```
Comment (npeltier) [inline libs/features/mas/web-components/src/variants/variant-layout.js:19]: no perf impact then?

File: libs/features/spectrum-web-components/dist/sidenav.js
Comment (npeltier) [inline libs/features/spectrum-web-components/dist/sidenav.js:1]: please ask for non regression on catalog

**Review by npeltier — APPROVED** [review]
looks good to me except preConnectedCallback hook that should be renamed (and both segment & mini compare cards tested)

File: libs/features/mas/web-components/src/variants/variant-layout.js (hunk 4 above)
Comment (yesil) [reply libs/features/mas/web-components/src/variants/variant-layout.js:19]: I need to measure :)

Comment (yesil) [reply libs/features/mas/web-components/src/variants/variant-layout.js:19]: will attach some data.

File: libs/features/spectrum-web-components/dist/sidenav.js
Comment (yesil) [reply libs/features/spectrum-web-components/dist/sidenav.js:1]: Sure, i think it is always par to of the validation. cc: @Roycethan @afmicka

File: libs/features/mas/web-components/src/merch-card.js (hunk 2 above)
Comment (yesil) [reply libs/features/mas/web-components/src/merch-card.js:110]: makes sense, I'll rename to `postUpdatedHook`

File: libs/features/mas/web-components/src/variants/variant-layout.js (hunk 4 above)
Comment (yesil) [reply libs/features/mas/web-components/src/variants/variant-layout.js:19]: it is slower :) will investigate further.

File: libs/features/mas/web-components/src/merch-card.js (hunk 2 above)
Comment (yesil) [reply libs/features/mas/web-components/src/merch-card.js:110]: Actually I'll move some logic to willUpdate hook, stay tuned.

**Comment by yesil** [comment]
Performance on the catalog page seems to be slightly positively impacted:

image

File: libs/blocks/merch-card/merch-card.js [hunk 5]
```diff
@@ -420,7 +420,7 @@ const addStartingAt = async (styles, merchCard) => {
   if (styles.includes('starting-at')) {
     const { replaceKey } = await import('../../features/placeholders.js');
     await replaceKey('starting-at', getConfig()).then((key) => {
-      const startingAt = createTag('div', { slot: 'starting-at' }, key);
+      const startingAt = createTag('div', { class: 'starting-at' }, key);
```
Comment (yesil) [inline libs/blocks/merch-card/merch-card.js:423]: updated needed after removing dead code in merch-card web component.

**Review by Roycethan — APPROVED** [review]

File: libs/features/mas/web-components/src/aem-datasource.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    /**
+     * Internal promise to track the readiness of the web-component to render.
+     */
+    _readyPromise;
+
+    static get observedAttributes() {
+        return ['path'];
+    }
+
+    attributeChangedCallback(name, oldValue, newValue) {
```
Comment (Axelcureno) [inline libs/features/mas/web-components/src/aem-datasource.js:78]: Looks like `oldValue` is not used

**Review by Axelcureno — APPROVED** [review]
Lgtm

**Pull request merged**